pkgs/development/python-modules/skein/skeinjar.nix

```diff
@@ -1,21 +1,19 @@
-{ callPackage, stdenv, maven, src, version }:
+{ fetchurl, unzip, stdenv, version, jarHash }:
 
-let
-  skeinRepo = callPackage ./skeinrepo.nix { inherit src version; };
-in
 stdenv.mkDerivation rec {
   pname = "skein.jar";
+  inherit version;
 
-  inherit version src;
+  src = fetchurl {
```

Comment:
Please use fetchPypi. Why can't we build from source?

Reply:
Can't use `fetchPypi` here. It only downloads the source code. We are building the python module from source in [skein/default.nix](https://github.com/NixOS/nixpkgs/pull/165226/files#diff-86ca46f8580a8a9246614cee5e1fdba5b0a78aef74ab8ae283fe9fa6b588d2fe). In this file [(skein/skeinjar.nix)](https://github.com/NixOS/nixpkgs/pull/165226/files#diff-6d9fd5cfe0ab3322d01a11c07142311d2199da43f77dac9ea78d95aa1e6f4dd5) we're only extracting a jar file from the prebuilt whl.

As Jon mentioned in https://github.com/NixOS/nixpkgs/commit/b9aacdf39e3c2075f7155ba87165e58933ba52ed, maven builds don't always produce the same hash. Until we have a better way of doing maven builds on nix, we'll have to use pebuilt jars.

Reply:
fetchPypi can also download wheels.

Reply:
I see... Updated.